javascript/ql/lib/semmle/javascript/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -196,6 +196,12 @@ class DataFlowCallable extends TDataFlowCallable {
 
   /** Gets the corresponding `LibraryCallable` if this is a library callable. */
   LibraryCallable asLibraryCallable() { this = MkLibraryCallable(result) }
+
+  int totalorder() {
+    result = TotalOrdering::astNodeId(this.asSourceCallable()).bitShiftLeft(1)
+    or
+    result = TotalOrdering::libraryCallableId(this.asLibraryCallable()).bitShiftLeft(1) + 1
+  }
```

Comment:
Why the left-shift?

Reply:
So the IDs for AST nodes and library callables can't clash, because the low bit is zero for AST nodes and 1 for library callables.